src/nvim/lua/executor.c

```diff
@@ -1605,9 +1611,10 @@ Object nlua_call_ref(LuaRef ref, const char *name, Array args, LuaRetMode mode,
     nlua_push_Object(lstate, &args.items[i], 0);
   }
 
-  if (nlua_pcall(lstate, nargs, 1)) {
-    // if err is passed, the caller will deal with the error.
-    if (err) {
+  bool fast = name && strcmp(name, "msg_show") == 0;
```

Comment:
To make this scalable we should add this as a flag to `api/ui_events.h` to mark the events which need this . I could implement that part myself on top of your work here.

Reply:
I took a stab at this because I first implemented the automatic detachment per UI extension where further information from the generator was necessary in C. Later decided that wasn't really necessary. Hope what I ended up with isn't too far off what you had in mind.